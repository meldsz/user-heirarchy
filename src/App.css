#root {
  padding: 15px;
}

.input-container {
  display: flex;
  gap: 10px;
  justify-content: flex-end;
  border: 2px;
}

.login-box {
  display: flex;
  flex-direction: column;
  gap: 15px;
  border: 2px solid purple;
  padding: 25px;
  justify-content: center;
}
.login-input {
  border: 2px solid purple;
  width: 300px;
}
.login-btn-container {
  display: flex;
  justify-content: flex-end;
}
.login-btn {
  background-color: #6c20aa;
  padding: 2px 25px;
}
.error-text {
  display: flex;
  justify-content: flex-end;
  color: red;
  width: 500px;
  margin: auto;
  padding: 10px;
}
.user-profile {
  position: absolute;
  right: 0;
  top: 0;
  padding: 10px;
  display: flex;
  gap: 5px;
}
.user-badge {
  height: 40px;
  width: 40px;
  border-radius: 50%;
  border: 2px solid purple;
  display: flex;
  align-items: center;
  justify-content: center;
}

.user-details {
  display: flex;
  flex-direction: column;
  gap: 5px;
  align-items: left;
  justify-content: center;
}

.user-children {
  padding-left: 40px;
  margin-left: 12px;
}

.user-container {
  display: flex;
  gap: 15px;
  padding: 10px;
}

.toggle {
  font-weight: 900;
  font-size: xx-large;
  padding-right: 10px;
  width: 15px;
  height: 10px;
  background: none;
  justify-content: center;
  display: flex;
  align-items: center;
  border: none;
  outline: none;
}

.not-found-container {
  width: 50%;
  margin: auto;
  line-height: 44px;
  text-align: center;
}

@media only screen and (min-width: 600px) {
  .login-box {
    padding: 25px;
    justify-content: center;
    width: 425px;
    margin: auto;
  }
  .user-details {
    display: flex;
    flex-direction: row;
    gap: 5px;
    align-items: center;
    justify-content: center;
  }
  .tree-container {
    padding-left: 40px;
  }
  .login-header {
    padding-left: calc(50% - 340px);
    padding-bottom: 50px;
  }
}

@media only screen and (max-width: 600px) {
  #root {
    width: 100%;
  }
}
@media only screen and (max-width: 480px) {
  .login-input {
    width: 150px;
  }
}
