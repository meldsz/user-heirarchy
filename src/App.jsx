import { UserContextProvider } from "./contexts/UserContext";
import { createBrowserRouter, RouterProvider } from "react-router";
import { routes } from "./common/routes";
import "./App.css";

function App() {
  const router = createBrowserRouter(routes);

  return (
    <UserContextProvider>
      <RouterProvider router={router} />
    </UserContextProvider>
  );
}

export default App;
