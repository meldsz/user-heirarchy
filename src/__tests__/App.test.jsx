import App from "../App";
import { render } from "@testing-library/react";

describe("<App/>", () => {
  it("renders the App component", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
