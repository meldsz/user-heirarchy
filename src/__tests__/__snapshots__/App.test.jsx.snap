// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<App/> renders the App component 1`] = `
<DocumentFragment>
  <div
    class="login-header"
  >
    <h1>
      Please Login
    </h1>
  </div>
  <div
    class="login-box"
  >
    <div
      class="input-container"
    >
      <span>
        email address:
      </span>
      <input
        aria-label="email address"
        class="login-input"
        data-testid="email"
        value=""
      />
    </div>
    <div
      class="input-container"
    >
      <span>
        password:
      </span>
      <input
        aria-label="password"
        class="login-input"
        data-testid="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="login-btn-container"
    >
      <button
        aria-label="Login"
        class="login-btn"
        data-testid="login-button"
      >
        Login
      </button>
    </div>
  </div>
</DocumentFragment>
`;
